rust/main/agents/relayer/src/msg/op_queue.rs

```diff
@@ -447,4 +489,114 @@ pub mod test {
         assert_eq!(popped[3], op_ids[0]);
         assert_eq!(popped[4], op_ids[1]);
     }
+
+    #[tracing_test::traced_test]
+    #[tokio::test]
+    async fn test_process_retry_requests_by_id() {
+        let (metrics, queue_metrics_label) = dummy_metrics_and_label();
+        let broadcaster = sync::broadcast::Sender::new(100);
+        let mut op_queue_1 = OpQueue::new(
+            metrics.clone(),
+            queue_metrics_label.clone(),
+            Arc::new(Mutex::new(broadcaster.subscribe())),
+        );
+
+        // Add some operations to the queue with increasing `next_attempt_after` values
+        let destination_domain: HyperlaneDomain = KnownHyperlaneDomain::Injective.into();
+        let messages_to_send = 5;
+        let ops: VecDeque<_> = (1..=messages_to_send)
+            .map(|seconds_to_next_attempt| {
+                Box::new(MockPendingOperation::new(
+                    seconds_to_next_attempt,
+                    destination_domain.clone(),
+                )) as QueueOperation
+            })
+            .collect();
+
+        let op_ids: Vec<_> = ops.iter().map(|op| op.id()).collect();
+
+        // push to queue 1
+        for op in ops {
+            op_queue_1
+                .push(op, Some(PendingOperationStatus::FirstPrepareAttempt))
+                .await;
+        }
+
+        let (transmitter, mut receiver) = mpsc::channel(ENDPOINT_MESSAGES_QUEUE_SIZE);
+
+        // Retry by message ids
+        broadcaster
+            .send(MessageRetryRequest {
+                uuid: "0e92ace7-ba5d-4a1f-8501-51b6d9d500cf".to_string(),
+                pattern: MatchingList::with_message_id(op_ids[1]),
+                transmitter: transmitter.clone(),
+            })
+            .unwrap();
+
+        op_queue_1.process_retry_requests().await;
+
+        let retry_response = receiver.recv().await.unwrap();
+
+        assert_eq!(retry_response.evaluated, 5);
+        assert_eq!(retry_response.matched, 1);
+    }
+
+    #[tracing_test::traced_test]
+    #[tokio::test]
+    async fn test_process_retry_requests_by_domain() {
```

Comment:
this test gives similar coverage to `test_destination_domain` with an additional check on the response content, so I think they can be merged